stac_fastapi/extensions/stac_fastapi/extensions/third_party/free_text/free_text.py

```diff
@@ -0,0 +1,46 @@
+"""Free-text extension."""
+
+from typing import List, Optional
+
+import attr
+from fastapi import FastAPI
+
+from stac_fastapi.types.extension import ApiExtension
+
+from .request import FreeTextExtensionGetRequest, FreeTextExtensionPostRequest
+
+
+@attr.s
+class FreeTextExtension(ApiExtension):
+    """Free-text Extension.
+
+    The Free-text extension adds an additional `q` parameter to `/search` requests which
+    allows the caller to perform free-text queries against STAC metadata.
+    https://github.com/stac-api-extensions/freetext-search/README.md
+    """
+
+    GET = FreeTextExtensionGetRequest
+    POST = FreeTextExtensionPostRequest
+
+    conformance_classes: List[str] = attr.ib(
+        factory=lambda: [
```

Comment:
As you advertise both basic and advanced at the same time, maybe we should first discuss https://github.com/stac-api-extensions/freetext-search/issues/10 ?